[#46] UserController과 관련 domain 코틀린 변경

msugo1 opened the pull request:
* UserController.kt

- UserController.java 클래스가 담고있던 메소드들을 코틀린으로 변경

Model 관련

* User.kt

- User.java의 내용을 User.kt로 변경

* UserFindPasswordRequest.kt

- UserFindPasswordRequest.java의 내용을UserFindPasswordRequest.kt로 변경

* UserLoginRequest.kt

- UserLoginRequest.java의 내용을 UserLoginRequest.kt로 변경

테스트 관련

* 코틀린으로 변경하면서 문제가 발생한 테스트 코드들 수정

Commit:
[#46] UserController과 관련 domain 코틀린 변경

* UserController.kt

- UserController.java 클래스가 담고있던 메소드들을 코틀린으로 변경

Model 관련

* User.kt

- User.java의 내용을 User.kt로 변경

* UserFindPasswordRequest.kt

- UserFindPasswordRequest.java의 내용을UserFindPasswordRequest.kt로 변경

* UserLoginRequest.kt

- UserLoginRequest.java의 내용을 UserLoginRequest.kt로 변경

테스트 관련

* 코틀린으로 변경하면서 문제가 발생한 테스트 코드들 수정

msugo1 commented:
일단 많이는 아니지만 지금까지 학습한 내용(책 + 공식문서)을 바탕으로 `UserController`과 `관련 클래스` 들만 우선적으로 코틀린으로 변경했습니다. 다른 코드들도 참고하면서 바꿔보긴 했는데, 코틀린이 아예 처음이라 한 번에 다 바꾸기보다는 코드를 작성하는 방향이 맞는지 검토를 먼저 받아보고 방향을 올바르게 잡는게 맞는 것 같아서 일단 UserController만 해서 PR 날려봅니다.

f-lab-dev commented on pom.xml:
```diff
@@ -129,6 +130,31 @@
+
+        <dependency>
+            <groupId>org.jetbrains.kotlin</groupId>
+            <artifactId>kotlin-stdlib-jdk8</artifactId>
```
이거는 위에서 선언한건데 중복선언 같습니다~

msugo1 replied:
auto configure을 잘못 눌러서 두 번 적용이 된 듯 합니다. 중복되는 부분 하나 삭제했습니다.

f-lab-dev commented on src/main/java/me/soo/helloworld/controller/UserController.kt:
```diff
@@ -0,0 +1,40 @@
+
+    private val userService: UserService,
+
+    private val loginService: LoginService) {
```
ktlint라는 것을 찾아서 한번 적용해보시면 좋을 것 같습니다~

msugo1 replied:
검색해보니 `https://github.com/pinterest/ktlint`가 나와서 여기 maven 관련 문서화 되어있는 내용 보고 적용했는데, 이게 의도하셨던바가 맞는가요? 의존성 추가하고 돌려보니 컨벤션 안맞는 부분은 잡아주긴 하더라구요.

+ git에 pre push 스크립트를 추가해서 push 전에 maven ktlint가 자동으로 컨벤션을 체크해주도록 설정했습니다

f-lab-dev replied:
네 맞습니다~ 좋네요 ㅎㅎ

f-lab-dev commented on src/main/java/me/soo/helloworld/controller/UserController.kt:
```diff
@@ -0,0 +1,40 @@
+
+    @GetMapping("/id-check")
+    fun checkDuplicateId(@RequestParam userId: String): HttpStatus {
+        return if (!userService.isUserIdExist(userId)) HttpStatus.OK else HttpStatus.CONFLICT
```
1. 이거도 `return`문을 생략 가능합니다
2. 이렇게 `HttpStatus` 객체를 반환하면 이 값은 헤더에 들어갈까요? 아니면 바디에 들어갈까요?

msugo1 replied:
1. Kotlin in Action 책에서 if 문도 식으로 쓸 수 있어 return을 생략할 수 있다던 부분이 생각났네요! 수정하겠습니다.

2. 본문에 들어갈 내용을 리턴하게 되니 Body에 들어가게 되네요. 실제로 아이디 중복 체크 후 문제가 있으면 헤더에 CONFLICT를 리턴하는 기존 테스트를 돌려보니 실패가 나오고, 포스트맨에서도 요청 보내보니 Body에 OK, CONFLICT 이런식으로 응답이 나오네요. 하나 더 배워갑니다.

f-lab-dev commented on src/main/java/me/soo/helloworld/model/user/User.kt:
```diff
@@ -0,0 +1,64 @@
+
+    val livingCountry: Int,
+
+    val livingTown: Int? = null,
```
default값을 주신 이유가 있으신가요?

msugo1 replied:
출신 국가(originCountry)/거주 국가(livingCountry)의 경우는 필수 입력사항인데, 거주 도시(livingTown)의 경우는 처음 설계를 했을 때 부터 선택 입력사항으로 두었기 때문에 값을 입력하지 않으면 null이 들어올 수 있어 null을 가능하게 해주는 연산자?를 추가하고, default 값을 null로 주었습니다.

f-lab-dev commented on src/main/java/me/soo/helloworld/model/user/User.kt:
```diff
@@ -0,0 +1,64 @@
+) {
+
+    fun createUserWithEncodedPassword(encodedPassword: String): User {
+        return User(
```
여기도 return이 생략 가능해보입니다~

msugo1 replied:
=을 이용해서 식으로 변경해서 return 생략했습니다

Commit:
[#46] 피드백 반영

* kotlin 의존성 중복선언 제거

* 컨벤션을 자동으로 체크해주는 ktlint 적용

* checkDuplicateId 메소드

- 함수를 식으로 바꾸어 리턴문 생략

- HttpStatus가 바디에 리턴되는 부분을 ResponseEntity를 추가하여 헤더에
  리턴될 수 있도록 변경

* User 클래스 내부에 있는 메소드인 `createUserWithEncodedPassword`
  메소드의 return을 생략하도록 변경

msugo1 committed:
Merge branch 'develop' into feature/46
